src/main/java/htsjdk/samtools/util/Histogram.java

```diff
@@ -153,12 +154,26 @@ public void increment(final K id, final double increment) {
     public void setValueLabel(final String valueLabel) { this.valueLabel = valueLabel; }
 
     /** Checks that the labels and values in the two histograms are identical. */
+    @Override
     public boolean equals(final Object o) {
+        if (o == this) {
+            return true;
+        }
         return o != null &&
                 (o instanceof Histogram) &&
                 ((Histogram) o).binLabel.equals(this.binLabel) &&
                 ((Histogram) o).valueLabel.equals(this.valueLabel) &&
-                super.equals(o);
+                ((Histogram) o).map.equals(this.map);
+    }
+
+    @Override
+    public String toString() {
+        return map.toString();
+    }
+
+    @Override
+    public int hashCode() {
+        return java.util.Objects.hash(binLabel, valueLabel, map);
```

Comment:
in case anyone wonders - Objects needs to be fully qualified here because it clashes with htsjdk.samtools.util.Objects (which should be removed btw)